### ISSUE TITLE
Nest non_fee_transfer under transaction

### ISSUE THREAD

**Issue by steven-sheehy** [hashgraph/hedera-mirror-node#6301]
### Problem

The transfers in the transaction body are the intended transfers sent by the client, but the transaction record produced by consensus nodes will aggregate the transfers and add in fee transfers. This makes it hard for users to map their transfers to what actually occurred. For allowances, the transfers in the body are the only ones with an `is_approval` flag and mapping that to aggregated transfers is not always possible. We have a `non_fee_transfer` table that is off by default to capture the intended transfers.

### Solution

* Add a `transaction.itemized_transfer` jsonb column
* Update `EntityRecordItemListener` to store the transfers from the body into the new field
* Migrate `non_fee_transfer` to `itemized_transfer`
* Remove `non_fee_transfer`

**Comment by mgoelswirlds** [comment]
@steven-sheehy Looking at the PubSubMessage, I see that we are setting nonFeeTransfers there as well. I believe we want to replace that with itemized transfers as well right?

**Comment by steven-sheehy** [comment]
Yes, but should keep the `PubSubMessage.nonFeeTransfers` named the same since it's part of the JSON doc sent to ETL.
